key-share/src/lib.rs

```diff
@@ -468,41 +475,51 @@ impl<E: Curve> AsRef<CoreKeyShare<E>> for CoreKeyShare<E> {
 }
 
 /// Error indicating that key share is not valid
-#[derive(Debug, thiserror::Error)]
-#[error(transparent)]
-pub struct InvalidCoreShare(#[from] InvalidShareReason);
+#[derive(Debug, displaydoc::Display)]
+#[cfg_attr(feature = "std", derive(thiserror::Error))]
+#[displaydoc("invalid core key share")]
+pub struct InvalidCoreShare(#[cfg_attr(feature = "std", source)] InvalidShareReason);
```

Comment:
Why change from transparent?

Reply:
I guess because a transparent error implies an incompatible display implementation?

Reply:
Transparent error will generate Display implementation no matter what. We want `displaydoc` to do Display impl, not `thiserror`